chatkit/src/main/java/com/stfalcon/chatkit/messages/MessageTextUtils.kt

```diff
@@ -29,36 +31,96 @@ class MessageTextUtils {
                 var isBold = false
                 var isItalic = false
                 var isStroke = false
-                when{
-                    group.startsWith("<") -> isLink = true
-                    group.startsWith("*") -> isBold = true
-                    group.startsWith("_") -> isItalic = true
-                    group.startsWith("~") -> isStroke = true
+                var surrounding = MarkDown.NONE
+                when {
+                    group.startsWith("<") -> {
+                        isLink = true
+                        surrounding = MarkDown.LINK
+                    }
+                    group.startsWith("*") -> {
+                        isBold = true
+                        surrounding = MarkDown.BOLD
+                    }
+                    group.startsWith("_") -> {
+                        isItalic = true
+                        surrounding = MarkDown.ITALIC
+                    }
+                    group.startsWith("~") -> {
+                        isStroke = true
+                        surrounding = MarkDown.STROKE
+                    }
                 }
-
                 group = group.substring(1, group.length - 1)
-                //w{"GROUP: $group"}
-
+                if(findNextMarkDown(0,group)!= -1) {
+                    var reqList = getTextPatterns(group)
+                    if(reqList.size > 0)
+                        reqList.forEach{unit: PatternDescriptor ->
+                            when{
+                                isBold == true -> unit.isBold = true
+                                isItalic == true -> unit.isItalic = true
+                                isStroke == true -> unit.isStroke = true
+                                isLink == true -> unit.isLink = true
+                            }
+                            list.add(unit)
+                        }
+                }
                 var url: PatternDescriptor? = null
                 if (group.contains("|") && isLink) {
                     val split = group.split("|")
                     if (split.isNotEmpty()) {
-                        url = PatternDescriptor(split[0], split[1], true, isBold, isItalic, isStroke)
+                        url = PatternDescriptor(removeMarkDowns(split[0]), removeMarkDowns(split[1]), true, isBold, isItalic, isStroke, surrounding)
                     }
                 } else {
-                    url = PatternDescriptor(group, null, isLink, isBold, isItalic, isStroke)
+                    url = PatternDescriptor(removeMarkDowns(group), null, isLink, isBold, isItalic, isStroke, surrounding)
                 }
                 url?.run {
-                    list.add(this)
+                    list?.add(this)
                 }
             }
-            //i{"FOUND: $list"}
             return list
         }
 
+        fun findNextMarkDown(currentIndex: Int, text: String) : Int
```

Comment:
can be private and please please please start bracket at function definition line